### PR TITLE
ATF Facade

### PR DESCRIPTION
**Pull request by aderiabin**
This proposal describes unified interface for ATF (Automated Test Framework) - ATF Facade.
ATF should provide unified and flexible interface for creation test scripts to perform checks of SDL functionality.

Main ATF Facade purposes:
 - Simplify access to ATF features for test scripts
 - Separate test scripts from ATF
 - Provide well-structured intuitive interface of ATF for test scripts
 - Unify way of use ATF for creating automated tests
 - Create single point of ATF features documentation
 - Provide ability to create more human-readable tests

### CODE REVIEW

**Comment by AStasiuk** [comment]
@theresalech proposal is ready for review

File: proposals/NNNN-atf_facade_proposal.md [hunk 1]
```diff
@@ -0,0 +1,255 @@
...
+## Introduction
+
+This proposal describes unified interface for ATF (Automated Test Framework) - ATF Facade.
+ATF should provide unified and flexible interface for creation test scripts to perform checks of SDL functionality.
+
+Main ATF Facade purposes:
+ - Simplify access to ATF features for test scripts
+ - Separate test scripts from ATF
+ - Provide well-structured intuitive interface of ATF for test scripts
+ - Unify way of use ATF for creating automated tests
```
Comment (theresalech) [inline proposals/NNNN-atf_facade_proposal.md]: Please update to "Unify way of using ATF for creating automated tests"

File: proposals/NNNN-atf_facade_proposal.md [hunk 2]
```diff
@@ -0,0 +1,255 @@
...
+ - Create single point of ATF features documentation
+ - Provide ability to create more human-readable tests
+
+## Motivation
+
+Currently ATF has a lot of functionality for testing most of SDL features, but each automated test uses different ATF modules.
+ATF does not have unified interface to provide its functionality for test scripts. Instead of this ATF has many separated modules with their own interfaces which are used in automated scripts for SDL testing purposes.
+
+So current approach leads to create test scripts which have multiple direct dependencies on different modules of ATF.
```
Comment (theresalech) [inline proposals/NNNN-atf_facade_proposal.md]: Please update to "So current approach leads to creating test scripts..."

File: proposals/NNNN-atf_facade_proposal.md [hunk 3]
```diff
@@ -0,0 +1,255 @@
...
+In other words automated test scripts are fully dependent on ATF implementation and each change in ATF can lead to need for change of test scripts.
```
Comment (theresalech) [inline proposals/NNNN-atf_facade_proposal.md]: Please update to "...can lead to a need to change test scripts."

File: proposals/NNNN-atf_facade_proposal.md [hunk 4]
```diff
@@ -0,0 +1,255 @@
...
+
+In addition current approach requires a strong knowledge in ATF modules purposes and interactions between them for create good automated test scenario using ATF.
```
Comment (theresalech) [inline proposals/NNNN-atf_facade_proposal.md]: Please update to "In addition, the current approach requires a strong knowledge in ATF modules' purposes and interactions between them in order to create a good automated test scenario using ATF."

File: proposals/NNNN-atf_facade_proposal.md [hunk 5]
```diff
@@ -0,0 +1,255 @@
...
+So creation of automated test scripts takes a lot of time and it can be a bit complicated for Mobile application developers in the beginning.
+
+Creation of ATF Facade will allow:
+ - to create test scripts which are independent of ATF implementation
+ - to unify ATF use approach
+ - to document ATF functionality
+ - to improve test scripts readability and their structure on base of scenarious
```
Comment (theresalech) [inline proposals/NNNN-atf_facade_proposal.md]: Please update "scenarios"

File: proposals/NNNN-atf_facade_proposal.md (hunk 1 above)
Comment (aderiabin) [reply proposals/NNNN-atf_facade_proposal.md]: Updated (see 9943914522e19530a121b117461502d1961c2b27)

File: proposals/NNNN-atf_facade_proposal.md (hunk 2 above)
Comment (aderiabin) [reply proposals/NNNN-atf_facade_proposal.md]: Updated (see 9943914522e19530a121b117461502d1961c2b27)

File: proposals/NNNN-atf_facade_proposal.md (hunk 3 above)
Comment (aderiabin) [reply proposals/NNNN-atf_facade_proposal.md]: Updated (see 9943914522e19530a121b117461502d1961c2b27)

File: proposals/NNNN-atf_facade_proposal.md (hunk 4 above)
Comment (aderiabin) [reply proposals/NNNN-atf_facade_proposal.md]: Updated (see 9943914522e19530a121b117461502d1961c2b27)

File: proposals/NNNN-atf_facade_proposal.md (hunk 5 above)
Comment (aderiabin) [reply proposals/NNNN-atf_facade_proposal.md]: Updated (see 9943914522e19530a121b117461502d1961c2b27)

**Comment by aderiabin** [comment]
@theresalech All comments above have been addressed. Thank you for review.

**Pull request merged**